tox.ini

```diff
@@ -1,29 +1,126 @@
+# TMPDIR, this will prefer putting those temp files in $XDG_RUNTIME_DIR,
+# which will speed up those tests since they'll run on a ramdisk.
+env_tmp_dir = {user_tmp_dir:{env:XDG_RUNTIME_DIR:{work_dir}}}/tox_tmp/{env_name}
+set_env =
+    TMPDIR={env_tmp_dir}
+    COVERAGE_FILE={env_tmp_dir}/.coverage_{env_name}
+
+[test]  # Base configuration for unit and integration tests
+package = editable
+extras = dev
+allowlist_externals = mkdir
+commands_pre = mkdir -p results
+
+[testenv:test-{py38,py39,py310,py311,py312}]  # Configuration for all tests using pytest
+base = testenv, test
+description = Run unit tests with pytest
+labels =
+    py38, py310, py311: tests, unit-tests
+commands = pytest {tty:--color=yes} --cov --cov-report=xml:results/coverage-{env_name}.xml --junit-xml=results/test-results-{env_name}.xml tests/unit {posargs}
+
+[testenv:integration-{py38,py39,py310,py311,py312}]
+base = testenv, test
+description = Run integration tests with pytest
+labels =
+    py38, py310, py311: tests, integration-tests
+commands = pytest {tty:--color=yes} --junit-xml=results/test-results-{env_name}.xml tests/integration {posargs}
+
+[lint]  # Standard linting configuration
+package = editable
+extras =
+    dev
+    test
+env_dir = {work_dir}/linting
+runner = ignore_env_name_mismatch
+
+[shellcheck]
+find = git ls-files
+filter = file --mime-type -Nnf- | grep shellscript | cut -f1 -d:
+
+[testenv:lint-{black,ruff,shellcheck,codespell,yaml}]
+description = Lint the source code
+base = testenv, lint
+labels = lint
+allowlist_externals =
+    shellcheck: bash, xargs
+commands_pre =
+    shellcheck: bash -c '{[shellcheck]find} | {[shellcheck]filter} > {env_tmp_dir}/shellcheck_files'
 commands =
-    pip install -U pip
-    pytest --basetemp={envtmpdir}
+    black: black --check --diff {tty:--color} {posargs} .
+    ruff: ruff --diff --respect-gitignore {posargs} .
+    shellcheck: xargs -ra {env_tmp_dir}/shellcheck_files shellcheck
+    codespell: codespell --toml {tox_root}/pyproject.toml {posargs}
+    yaml: yamllint {posargs} .
 
-[testenv:docs]
-commands = make docs
+[testenv:lint-{mypy,pyright}]
+description = Static type checking
+env_dir = {work_dir}/typing
+runner = ignore_env_name_mismatch
+package = editable
+extras = dev, types
+labels = lint, type
+allowlist_externals =
+    pyright: pyright
+    mypy: mkdir
+commands_pre =
+    mypy: mkdir -p .mypy_cache
+commands =
+    pyright: pyright {posargs}
+    mypy: mypy --install-types --non-interactive {posargs:.}
+
+[testenv:format-{black,ruff,codespell}]
+description = Automatically format source code
+base = testenv, lint
+labels = format
+commands =
+    black: black {tty:--color} {posargs} .
+    ruff: ruff --fix --respect-gitignore {posargs} .
+    codespell: codespell --toml {tox_root}/pyproject.toml --write-changes {posargs}
 
-[testenv:lint]
-commands = make lint
+[docs]  # Sphinx documentation configuration
+extras = doc
```

Comment:
starbase uses `docs`

Reply:
In this case shouldn't we also use `docs`?
I'll add Claudio to the review for his input

Reply:
I'm happy to use `docs` here or switch to `doc` for starbase. I changed it here so we had a minimal adjustment to craft-parts